src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationServiceImpl.java

```diff
@@ -66,6 +79,46 @@ public Integer process(MultipartFile document, Annotation annotation) {
         return annotationEntity.getId();
     }
 
+    @Override
+    public AnnotationResponseDto downloadAnnotationDoc(Integer annotationId, Integer annotationDocumentId) {
+
+        final Optional<ExternalObjectDirectoryEntity> dirEntity = eodRepository.findAnnotationIdAndAnnotationDocumentId(annotationId, annotationDocumentId);
+        final InputStreamResource stream;
+
+        final ExternalObjectDirectoryEntity externalObjectDirectoryEntity;
+
+        if (dirEntity.isEmpty()) {
+
+            if (authorisationApi.userHasOneOfRoles(List.of(SecurityRoleEnum.JUDGE))) {
+                throw new DartsApiException(INVALID_ANNOTATIONID_OR_ANNOTATION_DOCUMENTID_FOR_JUDGE);
+            } else {
+                throw new DartsApiException(INVALID_ANNOTATIONID_OR_ANNOTATION_DOCUMENTID);
+            }
+
+        }
+
+        externalObjectDirectoryEntity = dirEntity.get();
+
+        try {
+
+            stream = new InputStreamResource(dataManagementApi.getBlobDataFromInboundContainer(externalObjectDirectoryEntity.getExternalLocation()).toStream());
+
+        } catch (RuntimeException e) {
+            log.error("Failed to download annotation document {}", externalObjectDirectoryEntity.getExternalFileId(), e);
```

Comment:
ExternalFileId is normally blank, i think its only used my ARM. a better field to use might be the eod_id

Reply:
Done